src/ui/public/resize_checker/resize_checker.js

```diff
@@ -0,0 +1,94 @@
+import $ from 'jquery';
+import ResizeObserver from 'resize-observer-polyfill';
+import { uniqueId, isEqual } from 'lodash';
+
+import EventsProvider from 'ui/events';
+
+export function ResizeCheckerProvider(Private) {
+  const EventEmitter = Private(EventsProvider);
+
+  function validateElArg(el) {
+    // the ResizeChecker historically accepted jquery elements,
+    // so we wrap in jQuery then extract the element
+    const $el = $(el);
+
+    if ($el.size() !== 1) {
+      throw new TypeError('ResizeChecker must be constructed with a single DOM element.');
+    }
+
+    return $el.get(0);
+  }
+
+  function getSize(el) {
+    return [el.clientWidth, el.clientHeight];
```

Comment:
Could this trigger costly layout calculations? Why not use the values provided to the callback? http://rawgit.com/WICG/ResizeObserver/master/index.html#resizeobserverentry